### DISCUSSION TITLE
How to search for another (minimum) version than specified in CPMFindPackage()?

### DISCUSSION THREAD

**Discussion by bilke** [cpm-cmake/CPM.cmake#512] [Q&A]
I would like to use a CPM-installed dependency when the dependency is not found on the system so I use `CPMFindPackage()`. My project builds fine with an old version of the dependency so I would like that the find_package()-call behind the scenes uses an old version as the minimum. When not found I would like to let CPM download a new version of that dependency. Is that possible? Thanks!

**Comment by TheLartians** [comment]
I'm not 100% sure as it's been a while since writing the code, but I think we do forward the requested version to `find_package`, where CMake then decides if an available package is compatible or not (relevant [CMake docs](https://cmake.org/cmake/help/latest/command/find_package.html#version-selection)). If the version isn't found CPM should fall back to downloading the dependency from source.

**Reply by bilke** [reply]
Yes, but I would like to define two version: a minimum version for `find_package()` and  a download version for CPM.

**Reply by ClausKlein** [reply]
But may be esay possible with the current API:

```cmake
CPMFindPackage(
  GITHUB_REPOSITORY gabime/spdlog
  GIT_TAG v1.12.0 # for git clone if not found
  VERSION 1.10 # for find_package
  OPTIONS   "SPDLOG_INSTALL 1"
  SYSTEM ON
)
```
